### DISCUSSION TITLE
How to host the HTML package from Archi into a MS Power App ?

### DISCUSSION THREAD

**Discussion by cesgon** [yasenstar/ArchiSurance_Practice#1] [Q&A] [answered]
Perhaps this is outside your domain, but wondering if you know :)

**Comment by yasenstar** [comment, marked as answer]
Hello @cesgon ,

Thanks for your questions, it's really interesting to try publishing your Archi model to somewhere that others (non-models without Archi tool) can browse online.

Now you have the exported HTML package, you may see the root file is index.html, so that means you can use any kinds of web server to host your package, to my practice, I've create Node.js (actually Next.js) app and put the package under "/public/html" folder then the app can show that, and publish to one company internal web server, it's working.

I'd tried to utilize Microsoft Sharepoint to host the package, since that's the tool we internally doing the team collaboration, however, it's not convenient even it's doable, you can check this discussion in Archi forum: https://forum.archimatetool.com/index.php?topic=345.msg6764#msg6764, while, my showstopper was lack of permission and difficult to be granted to my managed Sharepoint site to do the customized iFrame editing.

According to Power platform, I know recently you can also have the way to use "Power Pages", but similar like SharePoint, Microsoft controls quite strictly and I can only see "iFrame" is the way you can connect to one foreign URL, but that means you should first host your HTML package is somewhere else (e.g. web server) and let Power Pages to access, so why do that additional step? And also, to myself, there's also not easy to get fully control permission with corporate network.

So, I suggest you start from the simpler approach, setup one normal web server, either publish direct to the webserver or create webapp and embed your package in. Good luck!

Xiaoqi

**Comment by cesgon** [comment]
Hey mate, thanks a lot
Really appreciate it all the details. at first I wanted in Sharepoint but
yes corporate IT rules are very strict. So I thought perhaps within MS
Power Apps Canvas.


Cesar Gonzalez MIEAust, TOGAF BA
Mobile +61 431 115 966
http://au.linkedin.com/in/cesaragonzalez
